### PR TITLE
Add Syllabus write page UI

### LINKED ISSUE #768: Syllabus: Create "Create a syllabus" page UI ("Write" tab, no functionality)
We need to create the "Create a syllabus" page as designed in Figma: https://www.figma.com/file/f1QMqiXSgt2FCt0itvsA24/676-Professor-Syllabus?node-id=162%3A1800

Screenshot 2022-02-23 at 18 32 03

This task only includes:

- [x] Creating the "Write" tab — https://github.com/Qiskit/platypus/pull/826

Out of scope for this task:

- The "Preview" tab
- State management of the forms (Vuex store)
- Functionalities like saving or updating the syllabus data
- Rich text (HTML) text fields that allow to enter bold text for example (instead, we'll only use regular text areas for now)

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes
Closes #768 

## Implementation details
- added Syllabus form
- added Syllabus Module block component
- added minimal/local state changes for updating strings per interaction
- leveraged `ColumnFlowGrid`

## How to read this PR
- This PR only covers the UI, and does not have full functionality

## TODO
- [x]  add `Add Content` cta that will add new Syllabus Module block to the DOM
- [x]  update functionality to restore `Save content` if course chapters are checked off
- [x]  delete Content block
- [x]  toast or notification for when content block has been deleted
- [x]  add cancel and publish buttons

## Screenshots

https://user-images.githubusercontent.com/6276074/159067441-898df6f2-caa9-4190-bb4a-62e4db6fea02.mov

### CODE REVIEW

File: frontend/vue/components/Syllabus/SyllabusFormModule.vue
Comment (korgan00) [inline frontend/vue/components/Syllabus/SyllabusFormModule.vue:139]: We can use our own color variables here

File: frontend/vue/components/Syllabus/SyllabusInlineNotification.vue
Comment (korgan00) [inline frontend/vue/components/Syllabus/SyllabusInlineNotification.vue:11]: Is this still needed?

File: frontend/vue/components/Syllabus/SyllabusInlineNotification.vue
Comment (korgan00) [inline frontend/vue/components/Syllabus/SyllabusInlineNotification.vue:65]: We can use our own color variables here

File: frontend/vue/main.ts
Comment (korgan00) [inline frontend/vue/main.ts:92]: I think that the form is not used directly in a `.pug` template or chapter, so is not needed here.

**Review by korgan00 — APPROVED** [review]
Great job, Randy!
I have some suggestions.
I didn't create a comment about translations, but there are many labels and texts without translation. We should review it before the feature merge, but I think the design is now stable. We can add it.

**Review by korgan00 — CHANGES_REQUESTED** [review]
I approved, sorry, i mean request changes

File: frontend/vue/components/Syllabus/SyllabusForm.vue
Comment (techtolentino) [reply frontend/vue/components/Syllabus/SyllabusForm.vue:30]: I think so, yes... but this can also come in a later PR.

Maybe @gracelindsell @pandasa123 might have input as well:

What should the user see if they delete multiple module/content blocks?

- 1 single inline notification telling the user that they deleted a Syllabus section/module
- multiple, stacked inline notifications telling the user that they deleted multiple Syllabus sections/modules
- something else

File: frontend/vue/components/Syllabus/SyllabusForm.vue
Comment (techtolentino) [reply frontend/vue/components/Syllabus/SyllabusForm.vue:37]: This is a question for @gracelindsell - thoughts on a confirmation modal, for when a user hits `Cancel`?

**Review by korgan00 — APPROVED** [review]
LGTM

**Pull request merged**
